### ISSUE TITLE
Build failure: clamav

### ISSUE THREAD

**Issue by MisileLab** [NixOS/nixpkgs#325510]
### Steps To Reproduce

1. build clamav

### Build log

https://hydra.nixos.org/build/264529717

### Notify maintainers

@robberer @qknight @globin

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.7, NixOS, 24.11 (Vicuna), 24.11.20240705.f252ef9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/xvvcqrx38fxyr7rc5pcljdxcyjv8kb7z-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
For ease of access, the error in question:
```
Connected to MAKE jobserver
    Start 1: libclamav
    Start 2: libclamav_rust
    Start 3: clamscan
    Start 4: clamd
    Start 5: sigtool
1/5 Test #3: clamscan .........................***Failed    0.43 sec

----------------------------------------------------------------------
Ran 0 tests in 0.000s

NO TESTS RAN

2/5 Test #5: sigtool ..........................   Passed    0.84 sec
3/5 Test #2: libclamav_rust ...................   Passed    8.40 sec
4/5 Test #1: libclamav ........................   Passed   10.56 sec
5/5 Test #4: clamd ............................   Passed   15.28 sec

80% tests passed, 1 tests failed out of 5

Total Test time (real) =  15.30 sec

The following tests FAILED:
          3 - clamscan (Failed)
Errors while running CTest
```

**Comment by KoviRobi** [comment]
I've done a bit of digging, it's failing with python3 unittest `_NO_TESTS_EXITCODE`.

I have run `nix develop nixpkgs#clamav`, `genericBuild`, and `checkPhase`, then started digging, in the build directory I saw `unit_tests/CTestTestfile.cmake` has the failing command
```
add_test(clamscan "/nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python3.12" "-m" "unittest" "--verbose" "clamscan")
```
and running that under `pdb`, I replaced `sys.exit` with a breakpoint and found it is doing `sys.exit(_NO_TESTS_EXITCODE)`.

So looks like `unittest` test discovery isn't picking up the files in https://github.com/Cisco-Talos/clamav/tree/main/unit_tests/clamscan.

Looking at https://github.com/Cisco-Talos/clamav/blob/main/CMakeLists.txt#L240 it seems like it first tries to use `pytest` then falls back on `unittest`, and indeed adding `pytest` to `checkInputs` seems to fix the failure.
